This notebook geocodes (translates from radar coordinates to lat/lon) the corner pixels of all my ROI's. 
1. Larger region (cropped_9000_1800_18000_3600)
2. Smaller region (cropped_9000_20_10000_1780)
3. cropped region at northern tip of Laguna Salada 

In [2]:
# import stuff
import os,sys
from osgeo import gdal

In [2]:
large_region_lat = '/data/ocp4/LagunaSalada/cropped_9000_1800_18000_3600/geom_reference/lat.rdr.full'
large_region_lon = '/data/ocp4/LagunaSalada/cropped_9000_1800_18000_3600/geom_reference/lon.rdr.full'
large_region_dx = 18000
large_region_dy = 3600

small_region_lat = '/data/ocp4/LagunaSalada/cropped_9000_20_10000_1780/geom_reference/lat.rdr.full'
small_region_lon = '/data/ocp4/LagunaSalada/cropped_9000_20_10000_1780/geom_reference/lon.rdr.full'
small_region_dx = 10000
small_region_dy = 1780

cropped_region_x1 = 14000
cropped_region_x2 = 17000
cropped_region_y1 = 1000
cropped_region_y2 = 1750

cropped_region_western_mtns_x1 = 14000
cropped_region_western_mtns_x2 = 18000
cropped_region_western_mtns_y1 = 1000
cropped_region_western_mtns_y2 = 2100

In [3]:
# large region 

# top right corner geographically 
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lat.rdr.full 0 0 
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lon.rdr.full 0 0 

# top left corner 
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lat.rdr.full 17999 0 
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lon.rdr.full 17999 0 

# bottom left corner
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lat.rdr.full 17999 3599 
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lon.rdr.full 17999 3599 

# bottom right corner
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lat.rdr.full 0 3599 
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lon.rdr.full 0 3599 


Report:
  Location: (0P,0L)
  Band 1:
    Value: 32.615454326997
Report:
  Location: (0P,0L)
  Band 1:
    Value: -115.055677567246
Report:
  Location: (17999P,0L)
  Band 1:
    Value: 32.7327620756761
Report:
  Location: (17999P,0L)
  Band 1:
    Value: -115.80872046165
Report:
  Location: (17999P,3599L)
  Band 1:
    Value: 32.2902303595692
Report:
  Location: (17999P,3599L)
  Band 1:
    Value: -115.924182661749
Report:
  Location: (0P,3599L)
  Band 1:
    Value: 32.169824434697
Report:
  Location: (0P,3599L)
  Band 1:
    Value: -115.155420271133


In [4]:
# small region 

# top right corner geographically 
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_20_10000_1780/geom_reference/lat.rdr.full 0 0 
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_20_10000_1780/geom_reference/lon.rdr.full 0 0 

# top left corner 
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_20_10000_1780/geom_reference/lat.rdr.full 9999 0 
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_20_10000_1780/geom_reference/lon.rdr.full 9999 0 

# bottom left corner
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_20_10000_1780/geom_reference/lat.rdr.full 9999 1779 
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_20_10000_1780/geom_reference/lon.rdr.full 9999 1779 

# bottom right corner
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_20_10000_1780/geom_reference/lat.rdr.full 0 1779 
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_20_10000_1780/geom_reference/lon.rdr.full 0 1779

Report:
  Location: (0P,0L)
  Band 1:
    Value: 32.8360065284444
Report:
  Location: (0P,0L)
  Band 1:
    Value: -115.007286694016
Report:
  Location: (9999P,0L)
  Band 1:
    Value: 32.9030092005015
Report:
  Location: (9999P,0L)
  Band 1:
    Value: -115.43488213151
Report:
  Location: (9999P,1779L)
  Band 1:
    Value: 32.6828729687555
Report:
  Location: (9999P,1779L)
  Band 1:
    Value: -115.48384819789
Report:
  Location: (0P,1779L)
  Band 1:
    Value: 32.6155786818662
Report:
  Location: (0P,1779L)
  Band 1:
    Value: -115.055653224566


In [6]:
# cropped region at tip of Laguna Salada

# top right corner geographically 
print('top right')
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lat.rdr.full 14000 1000 
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lon.rdr.full 14000 1000 

# top left corner 
print('top left')
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lat.rdr.full 16999 1000 
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lon.rdr.full 16999 1000 

# bottom left corner
print('bottom left')
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lat.rdr.full 16999 1749  
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lon.rdr.full 16999 1749 

# bottom right corner
print('bottom right')
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lat.rdr.full 14000 1749  
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lon.rdr.full 14000 1749 


top right
Report:
  Location: (14000P,1000L)
  Band 1:
    Value: 32.5844209517118
Report:
  Location: (14000P,1000L)
  Band 1:
    Value: -115.67545199537
top left
Report:
  Location: (16999P,1000L)
  Band 1:
    Value: 32.6031207418026
Report:
  Location: (16999P,1000L)
  Band 1:
    Value: -115.797126735641
bottom left
Report:
  Location: (16999P,1749L)
  Band 1:
    Value: 32.5103215459614
Report:
  Location: (16999P,1749L)
  Band 1:
    Value: -115.81660016082
bottom right
Report:
  Location: (14000P,1749L)
  Band 1:
    Value: 32.4916163493253
Report:
  Location: (14000P,1749L)
  Band 1:
    Value: -115.695055487993


In [7]:
# cropped region at Laguna Salada with western mtns 
cropped_region_western_mtns_x1 = 14000
cropped_region_western_mtns_x2 = 18000
cropped_region_western_mtns_y1 = 1000
cropped_region_western_mtns_y2 = 2100

# top right corner geographically 
print('top right')
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lat.rdr.full 14000 1000 
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lon.rdr.full 14000 1000 

# top left corner 
print('top left')
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lat.rdr.full 17999 1000 
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lon.rdr.full 17999 1000 

# bottom left corner
print('bottom left')
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lat.rdr.full 17999 2099  
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lon.rdr.full 17999 2099

# bottom right corner
print('bottom right')
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lat.rdr.full 14000 2099  
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lon.rdr.full 14000 2099 

top right
Report:
  Location: (14000P,1000L)
  Band 1:
    Value: 32.5844209517118
Report:
  Location: (14000P,1000L)
  Band 1:
    Value: -115.67545199537
top left
Report:
  Location: (17999P,1000L)
  Band 1:
    Value: 32.6091356230372
Report:
  Location: (17999P,1000L)
  Band 1:
    Value: -115.836432879008
bottom left
Report:
  Location: (17999P,2099L)
  Band 1:
    Value: 32.4736661354179
Report:
  Location: (17999P,2099L)
  Band 1:
    Value: -115.869488060882
bottom right
Report:
  Location: (14000P,2099L)
  Band 1:
    Value: 32.4482959068315
Report:
  Location: (14000P,2099L)
  Band 1:
    Value: -115.70451723996


In [6]:
# reference region I chose to use for interferograms
# used in 11_unwrapped_igrams/produce_weighted_disp_maps.ipynb for example

# top right corner geographically 
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lat.rdr.full 14800 1150 
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lon.rdr.full 14800 1150

# top left corner 
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lat.rdr.full 14899 1150 
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lon.rdr.full 14899 1150

# bottom left corner
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lat.rdr.full 14899 1179  
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lon.rdr.full 14899 1179 

# bottom right corner
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lat.rdr.full 14800 1179 
! gdallocationinfo /data/ocp4/LagunaSalada/Data_and_Figs/cropped_9000_1800_18000_3600/geom_reference/lon.rdr.full 14800 1179


Report:
  Location: (14800P,1150L)
  Band 1:
    Value: 32.571042497687
Report:
  Location: (14800P,1150L)
  Band 1:
    Value: -115.713176413358
Report:
  Location: (14899P,1150L)
  Band 1:
    Value: 32.5716276022211
Report:
  Location: (14899P,1150L)
  Band 1:
    Value: -115.716975783356
Report:
  Location: (14899P,1179L)
  Band 1:
    Value: 32.5680698888431
Report:
  Location: (14899P,1179L)
  Band 1:
    Value: -115.717965058226
Report:
  Location: (14800P,1179L)
  Band 1:
    Value: 32.5675024239414
Report:
  Location: (14800P,1179L)
  Band 1:
    Value: -115.714281140793


This is what a text file would look like 

# large subregion
32.615454326997     -115.055677567246       # top right 
32.7327620756761    -115.80872046165        # top left 
32.2902303595692    -115.924182661749       # bottom left 
32.169824434697     -115.155420271133       # bottom right 
32.615454326997     -115.055677567246       # top right 

# small subregion
32.8360065284444    -115.007286694016       # top right 
32.9030092005015    -115.43488213151        # top left 
32.6828729687555    -115.48384819789        # bottom left 
32.6155786818662    -115.055653224566       # bottom right 
32.8360065284444    -115.007286694016       # top right 

# cropped region from large subregion at tip of LS 
32.5844209517118    -115.67545199537        # top right 
32.6031207418026    -115.797126735641       # top left 
32.5103215459614    -115.81660016082        # bottom left 
32.4916163493253    -115.695055487993       # bottom right 
32.5844209517118    -115.67545199537        # top right 

# reference region within cropped region at tip of LS 
32.571042497687     -115.713176413358       # top right 
32.5716276022211    -115.716975783356       # top left 
32.5680698888431    -115.717965058226       # bottom left 
32.5675024239414    -115.714281140793       # bottom right 
32.571042497687     -115.713176413358       # top right 